libs/@local/hash-backend-utils/src/machine-actors.ts

```diff
@@ -228,64 +248,45 @@ export const createWebMachineActor = async (
   context: { graphApi: GraphApi },
   authentication: { actorId: ActorEntityUuid },
   {
-    webId,
+    shortname,
     logger,
     machineEntityTypeId,
   }: {
-    webId: WebId;
+    shortname: string;
     logger: Logger;
     machineEntityTypeId?: VersionedUrl;
   },
 ): Promise<ActorEntityUuid> => {
   const { graphApi } = context;
 
-  const machineAccountId = await graphApi
-    .createAccount(authentication.actorId, {
-      accountType: "machine",
-    })
-    .then((resp) => resp.data.id as ActorEntityUuid);
+  const systemAccountId = await graphApi
+    .getOrCreateSystemActor("h")
+    .then(({ data }) => data as MachineId);
 
-  await graphApi.modifyWebAuthorizationRelationships(authentication.actorId, [
-    {
-      operation: "create",
-      resource: webId,
-      relationAndSubject: {
-        subject: {
-          kind: "account",
-          subjectId: machineAccountId,
-        },
-        relation: "owner",
-      },
-    },
-  ]);
-
-  const systemAccountId = await getMachineActorId(context, authentication, {
-    identifier: "h",
-  });
+  const { webId, machineId } = await graphApi
+    .findWebByShortname(authentication.actorId, shortname)
+    .then(({ data }) => {
+      // eslint-disable-next-line @typescript-eslint/no-unnecessary-condition
+      if (!data) {
```

Comment:
Similar comment to elsewhere, can we make API types correct or throw an error instead?